src/db/graph/queries/get.rs

```diff
@@ -411,28 +414,125 @@ pub fn get_global_hot_tags_taggers(tag_list: &[&str]) -> Query {
     .param("labels", tag_list)
 }
 
-// Analyzes tag usage for a specific list of user IDs. Groups tags by name,
-// showing for each: label, post count and list of user IDs
-// Orders by post_count (descending).
-// Note: Only considers users from the provided users_id list.
-pub fn get_tags_by_user_ids(users_id: &[&str]) -> Query {
+fn tag_stream_reach_to_graph_subquery(reach: &TagStreamReach) -> String {
+    let query = match reach {
+        TagStreamReach::Followers => "MATCH (user:User)<-[:FOLLOWS]-(reach:User)",
+        TagStreamReach::Following => "MATCH (user:User)-[:FOLLOWS]->(reach:User)",
+        TagStreamReach::Friends => {
+            "MATCH (user:User)-[:FOLLOWS]->(reach:User), (user)<-[:FOLLOWS]-(reach)"
+        }
+    };
+    String::from(query)
+}
+
+pub fn get_tag_taggers_by_reach(
+    label: &str,
+    user_id: &str,
+    reach: TagStreamReach,
+    skip: usize,
+    limit: usize,
+) -> Query {
     query(
-        "
-        UNWIND $ids AS id
-        MATCH (u:User)-[tag:TAGGED]->(p:Post)
-        WHERE u.id = id
-        WITH tag.label AS label, COLLECT(DISTINCT u.id) AS taggers, COUNT(DISTINCT p) AS uniquePosts
-        WITH {
+        format!(
+            "
+            {}
+            MATCH (reach)-[tag:TAGGED]->()
+            WHERE user.id = $user_id AND tag.label = $label
+            WITH reach, MAX(tag.indexed_at) AS tag_time
```

Comment:
Please benchmark!
Do we need a new index for `indexed_at` on tag relationships?
```
"CREATE INDEX taggedIndexedAtIndex IF NOT EXISTS FOR ()-[r:TAGGED]-() ON (r.indexed_at)",
```    

Compare performance with and without index. Benchmarks will be useful.